src/V3Dfg.h

```diff
@@ -927,4 +869,48 @@ bool DfgVertex::isOnes() const {
     return false;
 }
 
+//------------------------------------------------------------------------------
+// Inline method definitions - for DfgGraph
+//------------------------------------------------------------------------------
+
+void DfgGraph::addVertex(DfgVertex& vtx) {
+    // Note: changes here need to be replicated in DfgGraph::addGraph
+    ++m_size;
+    if (DfgConst* const cVtxp = vtx.cast<DfgConst>()) {
+        m_constVertices.push_back(*cVtxp);
+    } else if (DfgVertexVar* const vVtxp = vtx.cast<DfgVertexVar>()) {
+        m_varVertices.push_back(*vVtxp);
+    } else {
+        m_opVertices.push_back(vtx);
+    }
+    vtx.m_userCnt = 0;
+    vtx.m_graphp = this;
+}
+
+void DfgGraph::removeVertex(DfgVertex& vtx) {
+    // Note: changes here need to be replicated in DfgGraph::addGraph
+    --m_size;
+    if (DfgConst* const cVtxp = vtx.cast<DfgConst>()) {
+        m_constVertices.erase(*cVtxp);
+    } else if (DfgVertexVar* const vVtxp = vtx.cast<DfgVertexVar>()) {
+        m_varVertices.erase(*vVtxp);
+    } else {
+        m_opVertices.erase(vtx);
+    }
+    vtx.m_userCnt = 0;
+    vtx.m_graphp = nullptr;
+}
+
+void DfgGraph::forEachVertex(std::function<void(DfgVertex&)> f) {
+    for (auto it = m_varVertices.begin(); it != m_varVertices.end();) f(*it++);
+    for (auto it = m_constVertices.begin(); it != m_constVertices.end();) f(*it++);
+    for (auto it = m_opVertices.begin(); it != m_opVertices.end();) f(*it++);
+}
+
+void DfgGraph::forEachVertex(std::function<void(const DfgVertex&)> f) const {
+    for (const DfgVertexVar& vtx : m_varVertices) f(vtx);
+    for (const DfgConst& vtx : m_constVertices) f(vtx);
+    for (const DfgVertex& vtx : m_opVertices) f(vtx);
+}
```

Comment:
And the usage is pretty simple (I hope). You use the iterator based traversal if you want to be able to remove the element being iterated on. Otherwise you can use range based loops.